bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/handlers/ParadoxAlarmHandlerFactory.java

```diff
@@ -41,6 +48,7 @@
 public class ParadoxAlarmHandlerFactory extends BaseThingHandlerFactory {
 
     private final Logger logger = LoggerFactory.getLogger(ParadoxAlarmHandlerFactory.class);
+    private final Map<ThingUID, @Nullable ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();

```

Comment:
Why final while it will then be updated ?

Reply:
Hm.... that really doesn't make any sense. Will change it.

Reply:
The map itself will be able to modify it's elements even though marked as final. We do not aim to change the pointer but probably from semantics perspective it's better to mark it as non-final...

Reply:
The map instance itself will never change, only the values inside, thus using final is okay.